mlptrain/potentials/mace/mace.py

```diff
@@ -0,0 +1,583 @@
+            data_loader = torch_geometric.dataloader.DataLoader(
+            dataset=[
+                data.AtomicData.from_config(config, z_table=z_table, cutoff=r_max)
+                for config in subset
+            ],
+            batch_size=valid_batch_size,
+            shuffle=False,
+            drop_last=False,
+            )
+
+            logging.info(f"Evaluating {name} ...")
+            _, metrics = evaluate(
+            model, loss_fn=loss_fn, data_loader=data_loader, device=device
+            )
+            if table_type == "TotalRMSE":
+                table.add_row(
+                [
+                    name,
+                    f"{metrics['rmse_e'] * 1000:.1f}",
+                    f"{metrics['rmse_f'] * 1000:.1f}",
+                    f"{metrics['rel_rmse_f']:.2f}",
+                ]
+                )
+            elif table_type == "PerAtomRMSE":
+                table.add_row(
+                [
+                    name,
+                    f"{metrics['rmse_e_per_atom'] * 1000:.1f}",
+                    f"{metrics['rmse_f'] * 1000:.1f}",
+                    f"{metrics['rel_rmse_f']:.2f}",
+                ]
+                )
+            elif table_type == "TotalMAE":
+                table.add_row(
+                [
+                    name,
+                    f"{metrics['mae_e'] * 1000:.1f}",
+                    f"{metrics['mae_f'] * 1000:.1f}",
+                    f"{metrics['rel_mae_f']:.2f}",
+                ]
+                )
+            elif table_type == "PerAtomMAE":
+                table.add_row(
+                [
+                    name,
+                    f"{metrics['mae_e_per_atom'] * 1000:.1f}",
+                    f"{metrics['mae_f'] * 1000:.1f}",
+                    f"{metrics['rel_mae_f']:.2f}",
+                ]
+                )
+        return table
+
+    @property
+    def requires_atomic_energies(self) -> bool:
+        return True
+
+    @property
+    def requires_non_zero_box_size(self) -> bool:
+        """ACE cannot use a zero size box"""
+        return True
+    
+    @property
+    def ase_calculator(self) -> 'ase.calculators.calculator.Calculator':
+        """ASE calculator for this potential"""
+  
+        try:
+            from mace.calculators import MACECalculator
+        except ModuleNotFoundError:
+            raise ModuleNotFoundError('MACE install not found, install it '
+                                      'here: https://github.com/ACEsuit/mace')
+        
+        calculator = MACECalculator(model_path = self.path+f"/checkpoints/{self.name}.model",
+                                   device='cpu', default_dtype="float64")
+        return calculator
+
+    def copy(self):
+        return deepcopy(self)
+    
+    @property
+    def filename(self):
+        return f'{self.name}.model'
+    
+    @property
+    def mp_start_method(self) -> float:
+    
+        return 'spawn'
+
+    def __init__(self,
```

Comment:
__inits__ are placed at the start of the class in mlptrain

Reply:
It doesn't matter right? some __init__ function for class of Molecule ... also placed at the end

Reply:
Molecule doesn't have an init class(?). All init functions in mlptrain are at the start. It doesn't make a difference functionality wise, but we should try to be consistent in how we write our code